[Text Directionality] Move the text directionality code to separate files

shallawa opened the pull request:
#### 50f1db9bfe6fdca4b08097845e0a0a6c73b5da44
<pre>
[Text Directionality] Move the text directionality code to separate files
<a href="https://bugs.webkit.org/show_bug.cgi?id=278698">https://bugs.webkit.org/show_bug.cgi?id=278698</a>
<a href="https://rdar.apple.com/134754984">rdar://134754984</a>

Reviewed by Ryosuke Niwa.

This is a follow-up of 282648@main. This will make the element text direction
code easier to read and debug.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Element.cpp:
(WebCore::parseTextDirectionState): Deleted.
(WebCore::elementTextDirectionState): Deleted.
(WebCore::elementHasValidTextDirectionState): Deleted.
(WebCore::elementHasAutoTextDirectionState): Deleted.
(WebCore::updateHasDirAutoFlagForSubtree): Deleted.
(WebCore::updateElementHasDirAutoFlag): Deleted.
(WebCore::computeTextDirectionFromText): Deleted.
(WebCore::computeTextDirection): Deleted.
(WebCore::computeContainedTextAutoDirection): Deleted.
(WebCore::computeTextDirectionOfSlotElement): Deleted.
(WebCore::updateEffectiveTextDirectionOfElementAndShadowTree): Deleted.
(WebCore::updateEffectiveTextDirectionState): Deleted.
(WebCore::Element::computeAutoTextDirection const): Deleted.
(WebCore::Element::computeTextDirectionIfDirIsAuto const): Deleted.
(WebCore::Element::updateEffectiveTextDirectionOfDescendants): Deleted.
(WebCore::Element::updateEffectiveTextDirectionOfAncestors): Deleted.
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementTextDirection.cpp: Added.
(WebCore::parseTextDirectionState):
(WebCore::elementTextDirectionState):
(WebCore::elementHasValidTextDirectionState):
(WebCore::elementHasAutoTextDirectionState):
(WebCore::updateHasDirAutoFlagForSubtree):
(WebCore::updateElementHasDirAutoFlag):
(WebCore::computeTextDirectionFromText):
(WebCore::computeTextDirection):
(WebCore::computeContainedTextAutoDirection):
(WebCore::computeTextDirectionOfSlotElement):
(WebCore::computeAutoDirectionality):
(WebCore::computeTextDirectionIfDirIsAuto):
(WebCore::updateEffectiveTextDirectionOfElementAndShadowTree):
(WebCore::updateEffectiveTextDirectionState):
(WebCore::updateEffectiveTextDirectionOfDescendants):
(WebCore::updateEffectiveTextDirectionOfAncestors):
* Source/WebCore/dom/ElementTextDirection.h: Added.
* Source/WebCore/html/HTMLTextFormControlElement.cpp:
(WebCore::HTMLTextFormControlElement::directionForFormData const):
* Source/WebCore/html/InputType.cpp:
(WebCore::InputType::setValue):
* Source/WebCore/style/ElementRuleCollector.cpp:
(WebCore::Style::ElementRuleCollector::matchAllRules):

Canonical link: <a href="https://commits.webkit.org/282796@main">https://commits.webkit.org/282796@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fec5453268ab9342cef0a8a1b1ec8d2f3f2c6d95

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/64272 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43629 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16869 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/68294 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14880 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/66392 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/51327 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/15160 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51724 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10261 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/67341 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40336 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55609 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32343 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37006 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12988 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13754 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/58969 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/13318 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69993 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/8219 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12836 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59046 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/8252 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55701 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/59208 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14191 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6792 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/476 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/39449 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/40528 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/41711 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/40271 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/dom/ElementTextDirection.cpp:
```diff
@@ -0,0 +1,314 @@
+}
+
+// Specs: https://html.spec.whatwg.org/multipage/dom.html#auto-directionality
+std::optional<TextDirection> computeAutoTextDirection(const Element& element)
```
We should really call this function computeAutoDirectionality to match the spec.

shallawa replied:
Done.

shallawa committed:
[Text Directionality] Move the text directionality code to separate files
https://bugs.webkit.org/show_bug.cgi?id=278698
rdar://134754984

Reviewed by Ryosuke Niwa.

This is a follow-up of 282648@main. This will make the element text direction
code easier to read and debug.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Element.cpp:
(WebCore::parseTextDirectionState): Deleted.
(WebCore::elementTextDirectionState): Deleted.
(WebCore::elementHasValidTextDirectionState): Deleted.
(WebCore::elementHasAutoTextDirectionState): Deleted.
(WebCore::updateHasDirAutoFlagForSubtree): Deleted.
(WebCore::updateElementHasDirAutoFlag): Deleted.
(WebCore::computeTextDirectionFromText): Deleted.
(WebCore::computeTextDirection): Deleted.
(WebCore::computeContainedTextAutoDirection): Deleted.
(WebCore::computeTextDirectionOfSlotElement): Deleted.
(WebCore::updateEffectiveTextDirectionOfElementAndShadowTree): Deleted.
(WebCore::updateEffectiveTextDirectionState): Deleted.
(WebCore::Element::computeAutoTextDirection const): Deleted.
(WebCore::Element::computeTextDirectionIfDirIsAuto const): Deleted.
(WebCore::Element::updateEffectiveTextDirectionOfDescendants): Deleted.
(WebCore::Element::updateEffectiveTextDirectionOfAncestors): Deleted.
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementTextDirection.cpp: Added.
(WebCore::parseTextDirectionState):
(WebCore::elementTextDirectionState):
(WebCore::elementHasValidTextDirectionState):
(WebCore::elementHasAutoTextDirectionState):
(WebCore::updateHasDirAutoFlagForSubtree):
(WebCore::updateElementHasDirAutoFlag):
(WebCore::computeTextDirectionFromText):
(WebCore::computeTextDirection):
(WebCore::computeContainedTextAutoDirection):
(WebCore::computeTextDirectionOfSlotElement):
(WebCore::computeAutoDirectionality):
(WebCore::computeTextDirectionIfDirIsAuto):
(WebCore::updateEffectiveTextDirectionOfElementAndShadowTree):
(WebCore::updateEffectiveTextDirectionState):
(WebCore::updateEffectiveTextDirectionOfDescendants):
(WebCore::updateEffectiveTextDirectionOfAncestors):
* Source/WebCore/dom/ElementTextDirection.h: Added.
* Source/WebCore/html/HTMLTextFormControlElement.cpp:
(WebCore::HTMLTextFormControlElement::directionForFormData const):
* Source/WebCore/html/InputType.cpp:
(WebCore::InputType::setValue):
* Source/WebCore/style/ElementRuleCollector.cpp:
(WebCore::Style::ElementRuleCollector::matchAllRules):

Canonical link: https://commits.webkit.org/282796@main
